File: frontend/src/app/services/movimiento.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Movimiento, MovimientoRequest } from '../models/movimiento.model';
import { environment } from '../../environments/environment';

@Injectable({ providedIn: 'root' })
export class MovimientoService {
  private readonly http = inject(HttpClient);
  private readonly url = `${environment.apiUrl}/movimientos`;

  listar(): Observable<Movimiento[]> {
    return this.http.get<Movimiento[]>(this.url);
  }

  buscarPorId(id: number): Observable<Movimiento> {
    return this.http.get<Movimiento>(`${this.url}/${id}`);
  }

  crear(request: MovimientoRequest): Observable<Movimiento> {
    return this.http.post<Movimiento>(this.url, request);
  }

  actualizar(id: number, request: MovimientoRequest): Observable<Movimiento> {
    return this.http.put<Movimiento>(`${this.url}/${id}`, request);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.url}/${id}`);
  }
}

File: backend/src/main/java/com/devsu/bank/service/MovimientoService.java
package com.devsu.bank.service;

import com.devsu.bank.dto.MovimientoRequest;
import com.devsu.bank.dto.MovimientoResponse;

import java.util.List;

public interface MovimientoService {
    MovimientoResponse crear(MovimientoRequest request);
    List<MovimientoResponse> listarTodos();
    MovimientoResponse buscarPorId(Long id);
    MovimientoResponse actualizar(Long id, MovimientoRequest request);
    MovimientoResponse actualizarParcial(Long id, MovimientoRequest request);
    void eliminar(Long id);
}
